misc-scripts/xref_mapping/XrefMapper/XrefLoader.pm

```diff
@@ -65,21 +65,18 @@ sub update{
   #####################################
   print "Deleting all PROJECTIONs from this database\n" if $verbose;
 
-  my $sql = "DELETE es FROM xref x, external_synonym es WHERE x.xref_id = es.xref_id and x.info_type = 'PROJECTION'";
+  my $sql = "TRUNCATE table ontology_xref";
   my $sth = $core_dbi->prepare($sql);
+  $sth->execute();
+  print "\tDeleted all ontology_xref rows\n" if $verbose;
+
+  $sql = "DELETE es FROM xref x, external_synonym es WHERE x.xref_id = es.xref_id and x.info_type = 'PROJECTION'";
+  $sth = $core_dbi->prepare($sql);
   my $affected_rows = $sth->execute();
   print "\tDeleted $affected_rows PROJECTED external_synonym row(s)\n" if $verbose;
-
-  # Delete all ontologies, as they are done by a separate pipeline
-  $sql = <<SQL;
-DELETE ontology_xref, object_xref, xref, dependent_xref
-FROM ontology_xref, object_xref, xref 
-LEFT JOIN dependent_xref on xref_id = dependent_xref_id
-WHERE ontology_xref.object_xref_id = object_xref.object_xref_id AND object_xref.xref_id = xref.xref_id
-SQL
+  $sql = "DELETE ox FROM object_xref ox, xref x, external_db e WHERE x.xref_id = ox.xref_id AND x.external_db_id =e.external_db_id and e.db_name = 'GO'";
```

Comment:
Is the intention to delete only from object_xref or from other tables too?
In the earlier version, alongwith ontology_xref, object_xref, xref and dependent_xref records were deleted too(?) for all ontologies.
Not sure if the other deletions(if still required) are handled in some other fashion.

Reply:
the GPAD loading pipeline, which now deals with updating the ontology_xref entries, has a clean up step at the start and end of the pipeline
the step here is redundant with the GPAD loading pipeline and quite heavy on the database